google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -313,4 +326,35 @@ public void testGetServiceAccountProjectId_nonExistentFile() throws Exception {
 
     assertNull(ServiceOptions.getServiceAccountProjectId(credentialsFile.getPath()));
   }
+
+  @Test
+  public void testResponseHeaderContainsMetaDataFlavor() throws Exception {  
+    Map<String, String> headers = new HashMap<String, String>();
+    HttpResponse httpResponse = createHttpResponseWithHeader(headers);
+    assertThat(ServiceOptions.headerContainsMetadataFlavor(httpResponse)).isFalse();
+
+    headers.put("Metadata-Flavor", "Google");
+    httpResponse = createHttpResponseWithHeader(headers);
+    assertThat(ServiceOptions.headerContainsMetadataFlavor(httpResponse)).isTrue();
+  }
+  
+  private HttpResponse createHttpResponseWithHeader(final Map<String, String> headers) throws Exception {
```

Comment:
I am not sure if `Map<String, String>` is the best practice here. `HttpResponse` seems to allow duplicate fields in some cases... Any suggestions?

Reply:
It does! If you wanted to allow for that, you could use a `Multimap` from Guava.